main.go

```diff
@@ -176,27 +177,30 @@ func main() {
 
 	logger.Info("starting UI...")
 
-	if !*noUI {
-		go func() {
-			<-done
-			exitErr(errors.New("exit with signal"))
-		}()
+	if *noUI {
+		<-done
+		return
+	}
 
-		if err := setupGUI(privateKey, messenger, logger); err != nil {
-			exitErr(err)
-		}
+	go func() {
+		<-done
+		exitErr(errors.New("exit with signal"))
+	}()
 
-		if err := messenger.Init(); err != nil {
-			exitErr(err)
-		}
+	if err := setupGUI(privateKey, messenger, logger); err != nil {
+		exitErr(err)
+	}
 
-		if err := g.MainLoop(); err != nil && err != gocui.ErrQuit {
-			exitErr(err)
-		}
+	if err := messenger.Init(); err != nil {
+		exitErr(err)
+	}
 
-		g.Close()
-	} else {
-		<-done
+	polling := startPolling(g, pollFunc, 50*time.Millisecond)
+	defer g.Close()
```

Comment:
Can we swap these two defers?

Not sure if `g.Close()` is needed because it is called in `exitErr`.

Reply:
I removed the `defer` and moved it to handle the situation where `err == nil || err == gocui.ErrQuit`. This is closer to what existed before. Good catch.